client/packages/glsp-sprotty/src/features/hints/element-type-hints-service.ts

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 2018 EclipseSource Services GmbH.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * 	Tobias Ortmayr - initial API and implementation
+ ******************************************************************************/
+import { injectable } from "inversify";
+import { SModelElement } from "sprotty/lib";
+import { contains } from "../../utils/array-utils";
+import { EdgeTypeHint, NodeTypeHint, SetTypeHintsAction, TypeHint } from "./action-definition";
+
+export interface TypeHintsService {
+    initialize(action: SetTypeHintsAction): void
+    isRepositionable(element: SModelElement): boolean
+    isDeletable(element: SModelElement): boolean
+    isResizeable(element: SModelElement): boolean
+
+    isContainer(element: SModelElement): boolean
+    isContainerChangeAllowed(newContainer: SModelElement, element: SModelElement): boolean
+
+    isRoutable(element: SModelElement): boolean
+    isAllowedEdgeType(source: SModelElement, target: SModelElement, edgeTypeId: string): boolean
+}
+
+@injectable()
+export class DefaultTypeHintsService implements TypeHintsService {
+    protected typeHints: Map<string, TypeHint> = new Map
+    protected nodeTypeHints: Map<string, NodeTypeHint> = new Map
+    protected edgeTypeHints: Map<string, EdgeTypeHint> = new Map
+    initialize(action: SetTypeHintsAction): void {
+        this.nodeTypeHints = new Map()
+        this.edgeTypeHints = new Map
```

Comment:
For the sake of consistency, it'd be better to be consistent with `new Map()` vs. `new Map`.